tests/utils/tensorflow/test_model_data_utils.py

```diff
@@ -22,14 +23,225 @@
     FEATURE_TYPE_SENTENCE,
     FEATURE_TYPE_SEQUENCE,
 )
-from rasa.utils.tensorflow.constants import SENTENCE
+from rasa.utils.tensorflow.constants import SENTENCE, SEQUENCE
 from rasa.shared.nlu.training_data.message import Message
 from rasa.shared.nlu.training_data.training_data import TrainingData
 from rasa.utils.tensorflow.model_data_utils import TAG_ID_ORIGIN
 
 shape = 100
 
 
+@pytest.mark.parametrize("seed", [123, 234])
+def test_combine_attribute_features_from_message_filters_correctly(seed: int,):
+    """Checks that we extract and combine the right features.
+
+    We check that we correctly filter
+    (1) which feature goes under which key in the output dictionary according to
+        whether it `is_sparse` and according to it's `type` and
+    (2) which features should be included according to the requested attribute and
+        list of featurizers,
+
+    Note that for (2) we create attributes and origin information at random such that
+    there are exactly two attributes and two featurizers used overall and we
+    request exactly of the two attributes and one of the two origins, respectively.
+    """
+    rng = np.random.default_rng(seed=seed)
+
+    # the total number of `Features` we want to create
+    sparse_type_combinations = list(
+        itertools.product([True, False], [SEQUENCE, SENTENCE])
+    )
+    num_feats_per_sparse_type_combination = 4
+    total_number_of_features = (
+        len(sparse_type_combinations) * num_feats_per_sparse_type_combination
+    )
+
+    # create 2 dummy attributes and featurizers ("origin")
+    attribute_flag = rng.choice(2, size=total_number_of_features)
+    featurizer_flag = rng.choice(2, size=total_number_of_features)
+    for flags in [attribute_flag, featurizer_flag]:
+        assert (
+            sum(flags) > 0 and sum(flags) < total_number_of_features
+        ), "Fix this test. Choose a different seed."

```

Comment:
I don't like this "choose a random seed so it works". How about this?
```suggestion
    num_ones = np.random.randint(1, 2 * total_number_of_features)
    random_flags = np.concatenate([
            np.ones(num_ones),
            np.zeros(2 * total_number_of_features - num_ones)
        ])
    np.random.shuffle(random_flags)
    attribute_flag, featurizer_flag = np.split(random_flags, 2)
```

Reply:
agree that this is not so nice - the suggested approach won't work but I can just "flip" the first element in an edge case (it won't matter)